app/assets/javascripts/channels/synchronized_editor_channel.js

```diff
@@ -23,13 +24,18 @@ $(document).on('turbolinks:load', function () {
 
         received(data) {
           // Called when there's incoming data on the websocket for this channel
-          if (current_user_id !== data['current_user_id']) {
-            CodeOceanEditor.applyChanges(data['delta']['data']);
+          if (data.command === 'editor_change' && current_user_id !== data['current_user_id']) {
+              CodeOceanEditor.applyChanges(data['delta']['data'], data['active_file']);
+          } else if (data.command === 'connection_change' && current_user_id !== data.current_user_id) {
+              CodeOceanEditor.showUserIsOnline(data);
+              this.perform('send_hello', {command: 'hello', status: 'connected', current_user_id: current_user_id, current_user_name: current_user_name});
```

Comment:
It feels like your `send_hello` method introduced in the backend is just a simple echo / forwarding, that would allow sending arbitrary data between the peers. While that works, for sure, it feels like most information is already present server-side, where we are also sure that it cannot be manipulated. Hence, I would add all this information just there...